src/point-cloud-layer.ts

```diff
@@ -0,0 +1,219 @@
+
+    /**
+     * The normal of each object, in `[nx, ny, nz]`.
+     * @default [0,0,1]
+     */
+    getNormal?: 10 // TODO normalAccessor
+    
+    /**
+     * The rgba color is in the format of `[r, g, b, [a]]`
+     * @default [0,0,0,225] 
+     */
+    getColor?: [0,0,0,225] // TODO colorAccessor
+}
+
+// Remove data nd get Position from the upstream default props
+const {
+    data: _data,
+    getPosition: _getPosition,
+    ..._upstreamDefaultProps
+} = PointCloudLayer.defaultProps;
+
+// Default props added by us
+const ourDefaultProps = {
+    _validate: true,
+};
+
+// @ts-expect-error Type error in merging default props with ours
+const defaultProps: DefaultProps<GeoArrowPointCloudLayerProps> = {
+    ..._upstreamDefaultProps,
+    ...ourDefaultProps,
+}
+
+export class GeoArrowPointCloudLayer<
+    ExtraProps extends {} = {},
+    > extends CompositeLayer<Required<GeoArrowPointCloudLayerProps> & ExtraProps>{
+        static defaultProps = defaultProps;
+        static layerName = "GeoArrowPointCloudLayer";
+
+        getPickingInfo(
+            params: GetPickingInfoParams & {
+                sourceLayer: { props: GeoArrowExtraPickingProps }
+            },
+        ): GeoArrowPickingInfo { 
+            return getPickingInfo(params, this.props.data);
+        }
+
+        renderLayers(): Layer<{}> | LayersList | null {
+            const { data: table } = this.props;
+        
+            const pointVector = getGeometryVector(table, EXTENSION_NAME.POINT);
+            if (pointVector !== null) {
+              return this._renderLayersPoint(pointVector);
+            }
+        
+            const geometryColumn = this.props.getPosition;
+            if (ga.vector.isPointVector(geometryColumn)) {
+              return this._renderLayersPoint(geometryColumn);
+            }
+        
+            throw new Error("geometryColumn not point or multipoint");
+        }
+
+        _renderLayersPoint(
+            geometryColumn: ga.vector.PointVector,
+        ): Layer<{}> | LayersList | null {
+            const { data: table } = this.props;
+            
+            if (this.props._validate) {
+                assert(ga.vector.isPointVector(geometryColumn));
+                validateAccessors(this.props, table);
+            }
+
+            // Exclude manually-set accessors
+            const [accessors, otherProps] = extractAccessorsFromProps(this.props, [
+                "getPosition",
+                "getNormal",
+                "getColors",
+            ]);
+            const tableOffsets = computeChunkOffsets(table.data);
+
+            const layers: PointCloudLayer[] = [];
+            for (
+                let recordBatchIdx = 0;
+                recordBatchIdx < table.batches.length;
+                recordBatchIdx++
+            ) {
+                const geometryData = geometryColumn.data[recordBatchIdx];
+                // is this actually three d coordinates data?
```

Comment:
A geoarrow point column can be 2 or 3 dimensional (or even 4). One thing we can do specifically for the point cloud layer is validate that the input is always 3d. So in the 
```js
if (this.props._validate)
```
block above, we can additionally add a check that the `type.child.listSize` (I think that's right from memory) is 3

Reply:
Addressed in PR